### PR TITLE
format codebase

### PR DESCRIPTION
**Pull request by MarinPostma**
More and more people are working on this codebase on a daily basis, is has become more convenient to adopt the rust style coding standard, with formatting provided by rustfmt, this ensure that we have a consistent style accross the codebase, and that contributors can  easily comply with it.

### CODE REVIEW

File: meilisearch-core/src/bucket_sort.rs [hunk 1]
```diff
@@ -559,15 +645,20 @@ impl<'txn> PostingsListView<'txn> {
 
     pub fn range(&self, range_offset: usize, range_len: usize) -> PostingsListView<'txn> {
         match self {
-            PostingsListView::Original { input, postings_list, offset, len } => {
+            PostingsListView::Original {
+                input,
+                postings_list,
+                offset,
+                len,
+            } => {
```
Comment (LegendreM) [inline meilisearch-core/src/bucket_sort.rs:653]: I prefer the previous state :(

**Review by LegendreM — APPROVED** [review]

File: meilisearch-core/src/bucket_sort.rs (hunk 1 above)
Comment (MarinPostma) [reply meilisearch-core/src/bucket_sort.rs:653]: Yeah it feels weird sometimes, but overall I hope it will force us to write nicer code.

Comment (MarinPostma) [reply meilisearch-core/src/bucket_sort.rs:653]: + Ra hints the types now, it can be useful

File: meilisearch-core/src/bucket_sort.rs [hunk 2]
```diff
@@ -405,12 +466,20 @@ fn cleanup_bare_matches<'tag, 'txn>(
...
+    for (
+        PostingsKey {
+            query,
+            input,
+            distance,
+            is_exact,
+        },
+        matches,
+    ) in queries
+    {
```
Comment (Kerollmops) [inline meilisearch-core/src/bucket_sort.rs:482]:
```suggestion
    for (postings_key, matches) in queries {
        let PostingsKey { query, input, distance, is_exact } = postings_key;
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Comment by qdequele** [comment]
2 things for me, I hope they have config for that. 
- Do not split macros arguments 
```rust
// keep
let path = concat!(env!("CARGO_MANIFEST_DIR"), "/../datasets/movies/movies.json");
// and not
let path = concat!(
    env!("CARGO_MANIFEST_DIR"), 
    "/../datasets/movies/movies.json"
);
```
- Do not split destructuring patern
```rust
// keep
let QueryResult { mut docids, queries } = traverse_query_tree(reader, &context, &operation)?;
// and not
let QueryResult {
    mut docids,
    queries,
} = traverse_query_tree(reader, &context, &operation)?;
```

**Comment by MarinPostma** [comment]
@qdequele I went throught [rustfmt's settings](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#configuration-file-resolution), and I think we can work out solutions. I'll try things out next week.

**Pull request closed without merging**